### ISSUE TITLE
Searchable Select field do not emit global change event

### ISSUE THREAD

**Issue by dalmarcolucas** [laravel/nova-issues#3569]
Laravel Version: 7.23.1
Nova Version: 3.29.0
PHP Version: 7.4.8
Database Driver & Version: MySql 8.0
Operating System and Version: Ubuntu 18.04.5 LTS
Browser type and version: 85.0.4183.121 (Official Build) (64-bit)
Reproduction Repository: /
Description:
First of all: Thank you for Nova!

I am working on a custom field that want's to listen to other fields and calculate things on the client side.
For that I looked into vue dev tools events and noticed that Nova conveniently emits events on every field natively.
So I thought that I could just listen to them. And that is partially true.
Since the Form Field emits global events through the handleChange() function I can listen to events like this:

Nova.$on("title-change", function (message) {
    console.log("title changed to: " + message);
});

But the SelectField, when using the option "searchable = true" does not emit this event globally. When not using the searchable option the event is emitted. 
Could you guys publish the event also when the searchable option is being used?

**Comment by dalmarcolucas** [comment]
Hi! I believe this is a BUG, not a request. Using the "searchable" option shouldn't change the behavior of publishing change events.

**Comment by crynobone** [comment]
Not a bug when it's intentionally designed not to publish it, you are exactly asking the feature to be added.

**Comment by dalmarcolucas** [comment]
> Not a bug when it's intentionally designed not to publish it, you are exactly asking the feature to be added.

Could you explain why is designed not to publish when the option searchable is true? I don't understand the logic

**Comment by crynobone** [comment]
> Could you explain why is designed not to publish when the option searchable is true? I don't understand the logic

A bug is where Nova uses such a feature and it doesn't work properly. When I read your description on the matter, you are asking for the feature to be added for your use case. If you want to file a bug report, please provide a full reproducible repo so we can review the code you used to trigger the bug as requested in the bug report template.

**Comment by dalmarcolucas** [comment]
@crynobone I think is easier just show you the code of the component:

In the file "Form/SelectField.vue" we have the handleChange func, who emit the change event:

![image](https://user-images.githubusercontent.com/17152587/135127996-957d1c9d-3546-4ad4-bee0-3ac1e9db4e2c.png)

It's called in the "change" event of the select-control (that is used when the option searchable is false):

![image](https://user-images.githubusercontent.com/17152587/135128146-8db5d228-f895-4ad5-b939-027d30a560aa.png)

But when the option searchable is true, it is not called:

![image](https://user-images.githubusercontent.com/17152587/135128247-97a0b1ce-a7d1-4a7d-beba-de95c929e018.png)

Basically, the solution would be just to emit the event in the "selectOption" that is the method that handle the change when using the searchable option:

![image](https://user-images.githubusercontent.com/17152587/135128692-e486246d-b26d-47cc-9697-0a73285e1418.png)
